tools/Google.Events.Tools.CodeGenerator/Program.cs

```diff
@@ -99,9 +99,18 @@ void GenerateCloudEventTypes()
                         writer.WriteLine("    {");
                         foreach (var type in group.Select(ce => ce.Type))
                         {
-                            var suffix = type.Split('.').Last();
+                            // Firestore and Datastore have separate "withAuthContext" variants; in those cases we want to
+                            // use the previous segment as well. Hard-coding this isn't pleasant, but it'll do until we see
+                            // anythign else we need to do.
+                            var suffix = type
+                                .Replace(".withAuthContext", "WithAuthContext")
+                                .Split('.')
+                                .Last();
+                            // "Undo" the hack before to get back to the relevant part of the type.
+                            // (We can't just use type.Split('.').Last(), as that would just give us "withAuthContext".)
+                            var description = suffix.Replace("WithAuthContext", ".withAuthContext");
```

Comment:
description is now `foo.withAuthContext`, just making sure that's intended.

Reply:
Yup, that's correct. (If you have a look in the context of the other constants, I think that's probably the best option.)